pkg/storage/stores/shipper/indexshipper/gatewayclient/gateway_client.go

```diff
@@ -184,93 +259,75 @@ func (s *GatewayClient) QueryIndex(_ context.Context, _ *logproto.QueryIndexRequ
 }
 
 func (s *GatewayClient) GetChunkRef(ctx context.Context, in *logproto.GetChunkRefRequest, opts ...grpc.CallOption) (*logproto.GetChunkRefResponse, error) {
-	if s.cfg.Mode == indexgateway.RingMode {
-		var (
-			resp *logproto.GetChunkRefResponse
-			err  error
-		)
-		err = s.ringModeDo(ctx, func(client logproto.IndexGatewayClient) error {
-			resp, err = client.GetChunkRef(ctx, in, opts...)
-			return err
-		})
-		return resp, err
-	}
-	return s.grpcClient.GetChunkRef(ctx, in, opts...)
+	var (
+		resp *logproto.GetChunkRefResponse
+		err  error
+	)
+	err = s.poolDo(ctx, func(client logproto.IndexGatewayClient) error {
+		resp, err = client.GetChunkRef(ctx, in, opts...)
+		return err
+	})
+	return resp, err
 }
 
 func (s *GatewayClient) GetSeries(ctx context.Context, in *logproto.GetSeriesRequest, opts ...grpc.CallOption) (*logproto.GetSeriesResponse, error) {
-	if s.cfg.Mode == indexgateway.RingMode {
-		var (
-			resp *logproto.GetSeriesResponse
-			err  error
-		)
-		err = s.ringModeDo(ctx, func(client logproto.IndexGatewayClient) error {
-			resp, err = client.GetSeries(ctx, in, opts...)
-			return err
-		})
-		return resp, err
-	}
-	return s.grpcClient.GetSeries(ctx, in, opts...)
+	var (
+		resp *logproto.GetSeriesResponse
+		err  error
+	)
+	err = s.poolDo(ctx, func(client logproto.IndexGatewayClient) error {
+		resp, err = client.GetSeries(ctx, in, opts...)
+		return err
+	})
+	return resp, err
```

Comment:
Nit: Could we change the signature of `poolDo` to accept a function that returns `(any, error)` instead of `error`?


```suggestion
	return s.poolDo(ctx, func(client logproto.IndexGatewayClient) (any, error) {
		return client.GetSeries(ctx, in, opts...)
	})
```